ha/setup/ha_setup.py

```diff
@@ -394,22 +441,30 @@ def process(self):
                     if node != node_name:
                         Log.info(f"Adding node {node} to Cluster {cluster_name}")
                         self._add_node(node, cluster_user, cluster_secret)
+                        # configure stonith
+                        configure_stonith(push=True, stonith_config=all_nodes_stonith_config.get(node))
             else:
                 # Add node with SSH
                 self._add_node_remotely(node_name, cluster_user, cluster_secret)
+                # configure stonith
+                configure_stonith(push=True, stonith_config=all_nodes_stonith_config.get(node_name))
         else:
+            # configure stonith for first node if cluster is already exists
+            configure_stonith(push=True, stonith_config=all_nodes_stonith_config.get(node_name))
             for node in nodelist:
                 if node != node_name:
                     Log.info(f"Adding node {node} to Cluster {cluster_name}")
                     self._add_node(node, cluster_user, cluster_secret)
+                    # configure stonith
+                    configure_stonith(push=True, stonith_config=all_nodes_stonith_config.get(node))
         self._execute.run_cmd(const.PCS_CLEANUP)
         Log.info("config command is successful")
 
-    def _create_resource(self, s3_instances, mgmt_info, node_count):
+    def _create_resource(self, s3_instances, mgmt_info, node_count, stonith_config=None):
         Log.info("Creating pacemaker resources")
         try:
             # TODO: create resource if not already exists.
-            create_all_resources(s3_instances=s3_instances, mgmt_info=mgmt_info, node_count=node_count)
+            create_all_resources(s3_instances=s3_instances, mgmt_info=mgmt_info, node_count=node_count, stonith_config=stonith_config)
```

Comment:
config should handle case if run multiple time

Reply:
We are handing the stonith resource is already present or not for that node in configure_stonith function.